From a053c2f68121f9234b8fb3d6f4a2ab7d6a294a56 Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Tue, 10 Jan 2017 15:44:55 +0100
Subject: [PATCH 0747/1345] ARM: mvebu: Add missing CIB register space for
 AXP/A370 pmsu

commit  b0ab910dd274665251c16cff3b2a7e28c47c9a83 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

Commit 8ab29bb80d2b ("ARM: mvebu: cpufreq: Protect entring WFI by
disabling the ACP port") added usage of second register space
in pmsu node (CIB - Control I/O Bridge). Due to driver dependency
on this register, Armada XP SoC was crashing very early during
boot.

This patch fixes above issue by adding CIB register description
in pmsu node in armada-370-xp.dtsi.

Partial fix 1/2 for JIRA SYSTEMSW-3238
("Armada XP fails to boot in the LSP v4.4.8")

Change-Id: Ib8913d8a42e52f7980db6205b15e17030263d889
Signed-off-by: Marcin Wojtas <mw@semihalf.com>
Signed-off-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/35868
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/armada-370-xp.dtsi |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/armada-370-xp.dtsi b/arch/arm/boot/dts/armada-370-xp.dtsi
index 383d92a..8b57647 100644
--- a/arch/arm/boot/dts/armada-370-xp.dtsi
+++ b/arch/arm/boot/dts/armada-370-xp.dtsi
@@ -225,7 +225,7 @@
 
 			pmsu: pmsu@22000 {
 				compatible = "marvell,armada-370-pmsu";
-				reg = <0x22000 0x1000>;
+				reg = <0x22000 0x1000>, <0x20280 0x4>;
 			};
 
 			usb0: usb@50000 {
-- 
1.7.9.5

